### PR TITLE
The Big View/Controller/Resource Rewrite

### PR DESCRIPTION
**Pull request by seancribbs**
This huge PR represents several weeks of trying to make the app and its resources more friendly to Ember.  Some things we did:
- The join-relationship between the `Test` and `Scorecard` entities is now enshrined in a `TestInstance` entity/model/resource. The models have been greatly simplified as a result.
- We don't create the matrix entries ahead of time (bye-bye `ScorecardSubcell`), but instead derive them from the contents of the scorecard's test instances. This is cleaner and results in slightly faster rendering of the matrix.
- The caching adapter is disabled for now, but the REST adapter now fetches in batches, reducing pauses and errors caused by URIs that are too long for the browser.
- Ember and Ember-Data have been updated.
- Lots of JS bugs/style issues have been fixed with `jshint`.

Miscellaneous other stuff:
- The font files are downloaded now with the proper media type.

CC: @cmeiklejohn @joedevivo @coderoshi

### CODE REVIEW

File: assets/javascripts/app/views.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 GiddyUp.prettyDate = function(date){
   // Based roughly on distance_of_time_in_words from Ruby on Rails,
   // with ideas from John Resig's "prettyDate" function.
+  if(date === null || date === undefined)
```
Comment (coderoshi) [inline assets/javascripts/app/views.js:4]: shouldn't that be `typeof(date) === 'undefined'`, or is that old school?

Comment (seancribbs) [inline assets/javascripts/app/views.js:4]: Actually `Ember.isNone(date)` would be better.

**Comment by coderoshi** [comment]
Awaiting the decrease in fetch batch size, and a reduction in requests, and presuming that the CachingAdapter isn't supposed to be used, +1

**Comment by seancribbs** [comment]
@coderoshi Thanks, merging and deploying. YAY

**Comment by cmeiklejohn** [comment]
Nice work, @seancribbs!

**Pull request merged**